pkg/cache/snapshot.go

```diff
@@ -79,6 +79,19 @@ func (s *Snapshot) Log(log logr.Logger) {
 			"usage", cohort.ResourceNode.Usage,
 		)
 	}
+
+	// Dump TAS snapshots if the feature is enabled
+	if features.Enabled(features.TopologyAwareScheduling) {
+		for cqName, cq := range s.ClusterQueues {
+			for tasFlavor, tasSnapshot := range cq.TASFlavors {
+				log.Info("TAS Snapshot Free Capacity",
+					"clusterQueue", cqName,
+					"resourceFlavor", tasFlavor,
+					"freeCapacityPerDomain", tasSnapshot.FreeCapacityPerDomain(),
```

Comment:
I would suggest the key is just "leafDomains". Then for each leaf domain we report:
- freeCapacity
- tasUsage

I think reporting nodeTaints would not hurt, but I', not sure it would also help much, so IMO we can skip it for now. cc @PBundyra

Reply:
WDYT about leaving freeCapacityPerDomain but then in the method `FreeCapacityPerDomain` iterate over all domains, not only leaves. I think it could be useful to have more comprehensive view into usage distribution.

Reply:
Yeah I meant the aggregated usage but I see it can be misleading because of resource fragmentation. In that case could we sort domains lexicographically so one can easily aggregate them themselves, wdyt?